acme/challenge.go

```diff
@@ -120,6 +125,19 @@ func (c Challenge) DNS01KeyAuthorization() string {
 	return base64.RawURLEncoding.EncodeToString(h[:])
 }
 
+// MailReply00KeyAuthorization encodes a key authorization value to be sent
+// back to the ACME challenge reply-to address.
+// Subject of that mail is separate token, token-part1,
+// and it's assembled with JSON token from challenge object, which rfc8823
+// calls token-part2
+// RFC8823 §3.1
+func (c Challenge) MailReply00KeyAuthorization(mailsubject string) string {
+	// if subject given has "ACME:" header, strip it before calculating the key authorization
+	mailsubject = strings.TrimPrefix(mailsubject, "ACME: ")
+	h := sha256.Sum256([]byte(mailsubject + c.KeyAuthorization))
```

Comment:
In my POC I've implemented base64-url decoding of the `token-part1` before combining the parts into a single key authorization (_not_ using this method). Assuming the server generates a random challenge (i.e. 32 characters), splits them in two, and then encodes `token-part1` using base64-url, as described in the RFC, I think that base64-url decoding should happen in this method before `token-part1` is prepended to the rest of the key authorization.

```text
The Subject header field has the following syntax: "ACME: <token-part1>", 
where the prefix "ACME:" is followed by folding white space (FWS; see [[RFC5322](https://datatracker.ietf.org/doc/html/rfc8823#RFC5322)]) 
and then by <token-part1>, which is the base64url-encoded first 
part of the ACME token that MUST be at least 128 bits long after decoding.
```

Reply:
yes I misread that part https://www.rfc-editor.org/rfc/rfc8823.html#section-3-6.6